src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7538,19 +7538,18 @@ private BoundExpression BindIndexedPropertyAccess(SyntaxNode syntax, BoundExpres
                     {
                         if (!candidate.IsStatic &&
                             candidate is PropertySymbol property &&
-                            property.GetOwnOrInheritedGetMethod() is MethodSymbol getMethod &&
-                            IsAccessible(getMethod, ref useSiteDiagnostics) &&
-                            getMethod.OriginalDefinition is var original &&
+                            IsAccessible(property, ref useSiteDiagnostics) &&
+                            property.OriginalDefinition is var original &&
```

Comment:
Do we have to check that the get accessor is accessible?

Reply:
That was actually a bug in my previous code -- normally when we bind an indexer or property access we don't check the accessor accessibility until later. Checking the `get` accessor will break if the indexer is only being used as a setter, and vice versa. The CheckValue method should now correctly figure out whether we need a getter or setter and provide an error if the necessary one is inaccessible. I have a test that reflects this in Semantics that